client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/GroupedInvestigations/GroupedInvestigationsForm/useGroupedInvestigationsForm.ts

```diff
@@ -0,0 +1,52 @@
+import axios from 'axios';
+import { useSelector } from 'react-redux';
+
+import logger from 'logger/logger'
+import { Severity } from 'models/Logger';
+import StoreStateType from 'redux/storeStateType';
+import useCustomSwal from 'commons/CustomSwal/useCustomSwal';
+
+export interface Reason {
+    id: number,
+    displayName: string
+}
+
+const useGroupedInvestigationsForm = ({ setReasons }: useGroupedInvestigationsFormInCome): useGroupedInvestigationsFormOutCome => {
+
+    const { alertError } = useCustomSwal();
+
+    const userId = useSelector<StoreStateType, string>(state => state.user.data.id);
+    const epidemiologyNumber = useSelector<StoreStateType, number>(state => state.investigation.epidemiologyNumber);
+
+    const fetchReasons = () => {
+        const reasonsLogger = logger.setup({
+            workflow: 'Fetching reasons for grouped investigations',
+            user: userId,
+            investigation: epidemiologyNumber
+        });
+        reasonsLogger.info('launching reasons request', Severity.LOW);
+        axios.get('/investigationInfo/groupedInvestigations/reasons')
+            .then((result: any) => {
+                result.data && setReasons(result.data);
```

Comment:
check result?.data. In addition this is an array, so you should check headers as well (you can see that on other axios.gets)

Reply:
In addition, don't you want to log  result.data invalid?

Reply:
@MayaSasson we don't do schemas on get requests anymore, so it will always have data and array

Reply:
fixed